### ISSUE TITLE
2024_10: BUG Sim UI PROD: team data files do not show after being uploaded

### ISSUE THREAD

**Issue by ljmoody** [lzim/teampsd#3216]
Tested in Edge and Chrome browsers.
VA laptop, VA login.

## Current behavior:
- inconsistent UX between Edge and Chrome browsers
- inconsistent UX within Chrome (between two test files)

### Test 1: Edge browser, team used to test: 3july_team_creation_team
1. From the Team Data window within the DDD, a user selects Upload to upload a file to the Sim:
![image](https://github.com/user-attachments/assets/32917bfb-664b-45ec-b99c-18228c2a9e2e)

2. I chose "data_to_test":
![image](https://github.com/user-attachments/assets/dce2b15d-0900-4b7d-b894-d0a10aee15a4)

3. The uploaded file does not appear in the Team Data file list:
![image](https://github.com/user-attachments/assets/b622e043-fae3-40bc-9e3c-87abff1ed8e0)

4. I refreshed the browser window; it's still not listed:
![image](https://github.com/user-attachments/assets/fe94629b-e8cc-414d-a8f3-4b35ed5311cb)

5. Hard reset (CTRL+F5); it's still not listed:
![image](https://github.com/user-attachments/assets/1c4938c5-2846-427a-8a5f-af2f98deddaf)

6. I selected Add and the file I just uploaded is not listed as a file to add (I did this twice):
![image](https://github.com/user-attachments/assets/f2238de7-aade-47ce-89cf-6d6518857307)

7. I logged off and back on again, went to the DDD, and now the file is showing in both the Session Start window and the Team Data file list:
![image](https://github.com/user-attachments/assets/38cd9199-0131-4763-9cf3-c9e37012540f)

### Test 2: Chrome browser, team used to test: 3july_team_creation_ind
1. Selected Upload to upload a file to the Sim.

2. Chose "data_to_test" file to upload; file successfully uploaded:
![image](https://github.com/user-attachments/assets/66a77979-a5d4-4e47-977d-ae30df4942a5)

3. Data file does not appear in Session Start nor Team Data windows:
![image](https://github.com/user-attachments/assets/a5df5d5d-c5f5-4702-b82a-3e519785247a)

4. I refreshed the browser window; now I can see the data file in Session Start and Team Data windows:
![image](https://github.com/user-attachments/assets/2e419f05-0a91-4ef8-bbf3-dc8772765005)
---
5. I decided to test via Chrome again, using a different data file - "blue_sample_data_flow":
![image](https://github.com/user-attachments/assets/92874f3d-0c7d-4388-bff0-e97e07e3fd56)

6. This data file was immediately seen in the Session Start and Team Data windows:
![image](https://github.com/user-attachments/assets/c9343b0b-e0b1-41c7-904a-a217d34a7c11)

### Test 3: Chrome browser, team used to test: 3july_team_creation_ind
1. From DDD, selected Upload > chose "2nd_sample_data_set" file, which uploaded successfully (missed getting a screenshot).
2. Data file showed immediately in both expected places:
![image](https://github.com/user-attachments/assets/45c51720-ed8a-4bf7-8de4-ac4c75b68f95)

### Another way users can get the team data file to appear:
1. After uploading a data file successfully, a user can select the Add button and associate the file with a team. I was unable to do this while testing because:
- In Test 1, step 6, I attempted this twice and it did not work. 
- My testing in Chrome didn't allow for the "Add" button to be pressed (the file appeared on the list after the first file was seen).  I have no more teams to test with today that aren't throwing errors reported in #3214 & #3203.  

## Desired behavior: 
- [ ] Uploading a team data file from the Team Data window in the DDD adds the file immediately to that window's data file list **and** the Session Start window.
- [ ] Users should not have to use the "Add" button, perform a browser refresh, nor log out and in again to view the file they uploaded.

**Comment by tejaspesquare** [comment]
@ljmoody 

Upon attempting to replicate this problem multiple times, we observed occasional inconsistencies. The file would appear in both sections, either upon refreshing or immediately after uploading it.
We will allocate time at a later date to identify the underlying cause. Hope this works for you.

**Comment by ljmoody** [comment]
@tejaspesquare, the issue is ongoing. This card was created when data files were not uploading. 
Today, the Admin Dashboard and DDD are not consistently handling upload or addition of Team Data files to teams. 
Documented in new issue #3220. These seem related.

**Comment by lzim** [comment]
@ljmoody 
@lijenn 

_How should we account for this is our scheduling of Consults?_

I plan to send invites tomorrow. LMK.
Thx!

@dlkibbe 
@dlounsbu

**Comment by lijenn** [comment]
@lzim I think it's okay to send out invites just to get the site meetings scheduled.

@dlkibbe and @dlounsbu are aware of the inconsistencies and are just focused on generating data ui files for the consults since @matthewtomo and I were unable to load their data files today ([#3220](https://github.com/lzim/teampsd/issues/3220#issuecomment-2243422870)), in addition to other bugs ([#3182](https://github.com/lzim/teampsd/issues/3182#issuecomment-2243457624), #3225).

But @tejaspesquare, _are there more updates to resolving these occasional inconsistencies in this card and #3220?_ 
We need to resolve these bugs ASAP.
@matthewtomo @jamesmrollins @ljmoody

**Comment by lzim** [comment]
Thanks for helping @dlkibbe & @dlounsbu understand how to move forward @lijenn !

**Comment by matthewtomo** [comment]
**Discussed support_workflow**
@ljmoody @lijenn @matthewtomo 

- It looks like this issue is more common on Edge, and that Chrome might be more stable
- We ran into issues replicating the bug on Edge. It occurred once, but we were unable to cause it again.
- We advise that it might be worth changing the text on the login screen to recommend users use Chrome for the Sim UI. 

![image](https://github.com/user-attachments/assets/4af4c2a7-479b-4489-b243-361dbe14ed32)

@lzim @jamesmrollins @tejaspesquare @dlounsbu @dlkibbe

**Comment by matthewtomo** [comment]
**Discussed 7/25 Standup with Dev**
@matthewtomo @tejaspesquare 

- DEV currently has someone working on recreating this issue. It is difficult because the issue is inconsistent. More effort will be allocated to this issue after resolving the team setup issues.
- Please record future occurrences of this bug here, so that we can continue attempting to diagnose causes.
- @lzim @jamesmrollins, would it be worth changing the text on the login screen to recommend users use Chrome for the Sim UI? (See above comment)

fyi:   @lijenn @ljmoody

**Comment by lzim** [comment]
@matthewtomo 

Yes, near term, I am okay with changing the message for now to help avoid errors that decrease use/engagement due to frustrating behavior. 

longer term, it would be better to find a more root cause solution than to require only Chrome.

> @lzim @jamesmrollins, would it be worth changing the text on the login screen to recommend users use Chrome for the Sim UI? (See above comment)

**Comment by lijenn** [comment]
Discussed at systems_sme/sim_ui/support sync:
@dlkibbe @dlounsbu @ljmoody @lijenn @matomasz 
- @dlkibbe got this same issue with the same team
- @lijenn was unable to recreate the issue
- @ljmoody @lijenn will try to have the console open and see if we can recreate the issue with @dl

**Comment by tejaspesquare** [comment]
Hello @ljmoody, @lijenn, @matomasz,

The problem has been resolved and is now ready for testing in the TEST environment. Kindly let us know if this issue is still ongoing.

**Comment by ljmoody** [comment]
fyi @tejaspesquare @lijenn @matthewtomo 

@ljmoody tested 8/9, VA laptop. There doesn't appear to be a long lag or any funny business with the data file upload happening like before. Looks good to go.

## Edge:
- added data_upload_test_file.xlsx to aug9_edge_3203_team
- file uploaded and appears in Team Data file list and in the Start a Session dropdown
✅ verified team data file upload is working correctly from the team Home Page. 

## Chrome:
- added second_test_data_set.xlsx to aug9_3203_teamleadassigned_team
- file uploaded and appears in Team Data file list and is available in the Start a Session dropdown
✅ verified team data file upload is working correctly from the team Home Page

**Comment by ljmoody** [comment]
**Discussed at 8/9/24 Team Praxis meeting:**
- We don't think the "Chrome only" login message is necessary. (see quoted comment below)
- Work is being done to ensure consistent UX throughout all browsers mentioned.
- Login browser message will stay as-is.

---
>- It looks like this issue is more common on Edge, and that Chrome might be more stable
>- We ran into issues replicating the bug on Edge. It occurred once, but we were unable to cause it again.
>- We advise that it might be worth changing the text on the login screen to recommend users use Chrome for the Sim UI. 
>
>![image](https://github.com/user-attachments/assets/4af4c2a7-479b-4489-b243-361dbe14ed32)

**Comment by matthewtomo** [comment]
wk3 2024_08 #sim_ui_workflow 
@matthewtomo @lzim @jamesmrollins @lijenn 

DECISION:
- [ ] @matthewtomo Use this text and log-in and close the card: "Please use Chrome, Edge, Firefox or Safari via desktop or mobile."
- consider a network quality message to prompt users that they may see latency.

**Comment by matthewtomo** [comment]
**wk3 2024_08 Dev Standup**
@matthewtomo @tejaspesquare

- [ ] @matthewtomo will handle the first bullet
- [ ] @matthewtomo will cover the second bullet on Friday Praxis meeting

**Comment by matthewtomo** [comment]
**week 3 2024_08 Praxis Meeting**
@matthewtomo @jamesmrollins @ljmoody 

> - [ ] @matthewtomo Use this text and log-in and close the card: "Please use Chrome, Edge, Firefox or Safari via desktop or mobile."

- Add a message above the file dropdown "Data is loading..." that clears once the file is loaded in
  - [ ] Message shows up in "Session Start" section
  - [ ] Message shows up in "Team Data" section

**Comment by matthewtomo** [comment]
- All final discussed behaviors have been implemented in DEV environment.
- Propose issue remains open until changes are deployed into the PROD environment
> * [x] Updated login message to: "Please use Chrome, Edge, Firefox or Safari via desktop or mobile."
![image](https://github.com/user-attachments/assets/1a0bd49a-9341-4da4-b954-22be151db943)
> * Add a message above the file dropdown "Data is loading..." that clears once the file is loaded in (show at the same time as the 3 loading bars)
>  
>   * [x] Message shows up in "Session Start" section
>   * [x] Message shows up in "Team Data" section
![image](https://github.com/user-attachments/assets/71f57f02-53df-495f-b0bc-b9bd77acdf3b)

**Comment by lzim** [comment]
@matthewtomo 
_Is this ready to close?_

This has been deployed to PROD by now, right?
Thx!

**Comment by matthewtomo** [comment]
@lzim 

The main issue of this card was resolved, and behavior changes were deployed into PROD, but there was one final discussed bullet to be addressed (better indicators that the data is still loading). 

The behaviors for that bullet are a part of the current slate of updates being tested in the TEST environment. In order to best track these new updates being tested, I would prefer to wait on closing this card until the changes for that last discussed bullet are live and confirmed in PROD .

**Comment by lzim** [comment]
thanks @matthewtomo

**Comment by ljmoody** [comment]
Verified behavior in Edge and Chrome 10/9/2024 (TEST instance, VA laptop) for documentation purposes.

✅ login screen message observed:  "Please use Chrome, Edge, Firefox or Safari via desktop or mobile."
✅ "Data is uploading..." message observed in both the Session Start and Team Data windows:
![image](https://github.com/user-attachments/assets/fb43fd33-ae89-4595-b063-fe76c386dd2d)

- Moving this card to Done pipelines. 
- Card will be closed after updates are pushed to PROD.

fyi @matthewtomo @tejaspesquare

**Comment by matthewtomo** [comment]
Verified behavior in Chrome 11/22/2024 (PROD instance, non-VA laptop).

✅ login screen message observed:  "Please use Chrome, Edge, Firefox or Safari via desktop or mobile."
![image](https://github.com/user-attachments/assets/f41f0613-61db-4b4e-a409-7e131421c96f)

✅ "Data is uploading..." message observed in both the Session Start and Team Data windows:
![image](https://github.com/user-attachments/assets/e6e9dae3-ab3f-4175-9056-467fb49b1277)

Behavior is correct and in PROD. Marking card as closed.
